### ISSUE TITLE
In flutter 3.10 our app crash when user start typing in text field in some windows devices

### ISSUE THREAD

**Issue by iampopal** [flutter/flutter#127789]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

This issue happens on some windows 10 devices only. 
This issue does not exist in version 3.7.3 of flutter we downgrade to flutter 3.73 the issue is solved.

### Expected results

The app crash when user start typing in TextField

### Actual results

App Crashes

### Code sample

<details><summary>Code sample</summary>

```dart
Just a simple TextField with a Controller
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
This issue only Happens and some of Windows 10 devices of our clients and we didn't able to reproduce in our own computers so want flutter team to please test flutter 3.10.2 in multiple windows devices 
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
The issue happens in flutter 3.10 and above the issue not exists in 3.7.3 and we rolled back
```

</details>

**Comment by huycozy** [comment]
Hi @iampopal 
The issue should be actionable with clear steps to reproduce so that we can address the issue properly. In case it's only happening in production, please try to ask for the stack trace along with device information from your clients.

**Comment by iampopal** [comment]
The issue is happening only in 10% of windows computers of our clients

We have downgraded to flutter 3.7.3 to solve this issue.

The issue is: the app crush when user start typing

Can you please tell us how we can get stacktrace of user’s computer when
they start typing the app crush?

**Comment by huycozy** [comment]
Unfortunately, I have no idea about this scenario when the issue happens on your client machine.

Does your project use any crash analytic services? If so, please try to get the crash log and share it here so we can investigate the issue. 

You also can try running your current project in release mode and check if the crash can occur. It would be better if you have your client's device info and try to set up the same environment to reproduce the issue.

**Comment by iampopal** [comment]
The issue only happens on couple windows 10 devices

I have tried to installed the same Windows 10 version but didn’t got that
error message on my local machine.

I think it’s causing due to some hardware or graphics issue of client
computer.

About crashlytics:
I have looked on:
firebase_crashlytics
But it’s not available for windows,

can you please help how to get user windows app crash and write it into our
server so we can understand and get the error.

**Comment by huycozy** [comment]
@iampopal Without additional information, we are unfortunately not sure how to resolve this issue.

We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the issue if you have any more information for us, and then we will re-open issue for further investigation! Thank you!

**Comment by luwei888** [comment]
I had the same problem with Windows 11

**Comment by luwei888** [comment]
![5e6777df66cb2feafce8f8d6d03fb47](https://github.com/flutter/flutter/assets/47804615/7ccfc05e-896e-461e-8c13-df100c74a0e9)

**Comment by iampopal** [comment]
@luwei888 thank you so much for sharing this issue error and screenshot. 

@huycozy we would you reopen this issue and find a solution for this issue ASAP so we could update to new version of flutter.

**Comment by luwei888** [comment]
1.focus to field
2.change field Controller.text

**Comment by luwei888** [comment]
@huycozy
1.focus to field
2.change field Controller.text will repeat this question

**Comment by huycozy** [comment]
@luwei888 Can you please provide a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) along with reproduction steps?

Also, please provide the output of `flutter doctor -v` and  `flutter run -v` that may contain the crash stack trace. The output log should be in text format instead of a screenshot for better readability and investigation. 

You may want to refer to [writing and formatting syntax](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) for beautifully formatted issue description. Thank you!

**Comment by iampopal** [comment]
@huycozy this issue happens whenever use: 

1.focus to field
2.change field Controller.text will repeat this question

please try it with any example of flutter 3.10.+ this issue will repeat in windows 10 and mostly windows 11 as @luwei888  showed error of this too.

**Comment by huycozy** [comment]
>change field Controller.text will repeat this question

@iampopal Can you specify this step? Is that reproducible on your side?
I try below sample code on my Windows but can't see the issue:

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  
  final tc = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        body: Center(
          child: TextField(
            controller: tc,
          ),
        ),
      ),
    );
  }
}
```
</details>

>as @luwei888 showed error of this too.

Unfortunately, we need a complete stack trace in text format that will be more useful to investigate this issue.

**Comment by iampopal** [comment]
1) Use a focusNode with it and make a button which will call:
focusNode.requestFocus()

2) use textController.text = “new Text”

As @luwei888 <https://github.com/luwei888> said this issue raising when
setting text of textController and when changing focus of focus Node.

This bug is not happening to me on my pc but happening in other users pc’s
which are more then 100 users.

I am unable to reach them so i downgraded to flutter 3.7 and my issue is
solved.

The screenshot of @luwei888 <https://github.com/luwei888> is right, i
believe its an issue coming from c++ side as the error show.

**Comment by liangjiate** [comment]
```
import 'dart:async';
import 'dart:math';

import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final tc = TextEditingController(text: "old text");

  final fc = FocusNode();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        body: Center(
          child: Column(
            children: [
              TextField(
                controller: tc,
                focusNode: fc,
              ),
              TextButton(
                onPressed: () {
                  Timer.periodic(const Duration(seconds: 1), (timer){
                    fc.requestFocus();
                    var randomInt = Random().nextInt(1000);
                    tc.text = "new $randomInt text";
                  });
                },
                child: const Text("test"),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

Flutter 3.10.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 682aa387cf (3 days ago) • 2023-06-05 18:04:56 -0500
Engine • revision 2a3401c9bb
Tools • Dart 3.0.3 • DevTools 2.23.1

Running flutter doctor...
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.4, on Microsoft Windows [版本 10.0.22000.1936], locale zh-CN)
[✓] Windows Version (Installed version of Windows is version 10 or higher)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Chrome - develop for the web
[✓] Visual Studio - develop for Windows (Visual Studio Professional 2022 17.5.2)
[✓] Android Studio (version 2022.2)
[✓] Connected device (3 available)

you can try these code @huycozy . thank you .

**Comment by liangjiate** [comment]
I try flutter run -v  , but can't see any useful information

[  +26 ms] The Flutter DevTools debugger and profiler on Windows is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:51511/9yfIEJV56Ag=/
[+7787 ms] Service protocol connection closed.
[   +2 ms] Lost connection to device.
[   +1 ms] DevFS: Deleting filesystem on the device (file:///C:/Users/luwei/AppData/Local/Temp/textfiledd55c86/textfile/)
[ +257 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed

[   +3 ms] "flutter run" took 38,759ms.
[ +239 ms] ensureAnalyticsSent: 233ms
[        ] Running 1 shutdown hook
[   +9 ms] Shutdown hooks complete
[        ] exiting with code 0

**Comment by liangjiate** [comment]
@huycozy  What is your version of Visual Studio?

> Thank @liangjiate for the sample code. But unfortunately, the issue can not reproduce on my machine. Maybe it only appears on some specific machines as @iampopal mentioned above.
> 
> Since you can reproduce it locally, can you try debugging it on Visual Studio to get the crash log for investigating this? (You can check [Compiling with Visual Studio](https://docs.flutter.dev/platform-integration/windows/building#compiling-with-visual-studio) as a guideline for this)

**Comment by iampopal** [comment]
@liangjiate please do try to run this simple on Visual Studio code and share the crash video if possible, because we need a solution for this because of this issue our app is still using an older version of Flutter 3.7

@huycozy I think the only way to find this issue is to run an app in multiple windows 10 and windows 11 computers I have 3 PCS it's not happening in any one of them but 20% of our users are facing this issue. 

waiting for a quick fix for this issue.

**Comment by luwei888** [comment]
https://github.com/flutter/flutter/assets/47804615/6509f7b3-b008-49fa-97c9-0af1e9af2e58

**Comment by luwei888** [comment]
> 20230619_103504.mp4
@huycozy I don't know how to debug in flutter window.dll. Can you tell me?

**Comment by huycozy** [comment]
@iampopal Unfortunately, I don't have many Windows devices to check this.

Thank @luwei888 for the reproduction. Even though I can not replicate this issue, perhaps the stack trace above is useful for others in order to investigate this further.

Can you also provide the complete output of `flutter doctor -v` which contains the Visual Studio version? 

Also, this may be related to an old issue: https://github.com/flutter/flutter/issues/116753. Have you modified Windows native code?

**Comment by yaakovschectman** [comment]
I can repro on _one_ Windows device, but for some reason VS on that device is giving me a real hard time on debugging. It seems `GetFlutterPlatformNodeDelegateFromID(ax_node->id()).lock()` is returning null

**Comment by iampopal** [comment]
@yaakovschectman thank you for your help, we are still waiting for a fix of this and still didn't able to release new version of our app

**Comment by yaakovschectman** [comment]
We've merged the fix into the main branch of the engine. Check that using the master channel no longer reproduces the bug.

**Comment by luwei888** [comment]
This issue is fixed in 3.10.6

**Comment by iampopal** [comment]
Gloat hearing that the issue is fixed, will be waiting for stable release
to test it. Thank your all your contributions.
